servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/ReadOnlyContextMap.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.internal;
+
+import io.servicetalk.context.api.ContextMap;
+
+import java.util.Map;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+public final class ReadOnlyContextMap implements ContextMap {
```

Comment:
Bcz initially it's used only in one place, consider starting with a pkg-private class under http-api package